src/plugins/blocksmith/components/pinterest-pin-block-modal/index.jsx

```diff
@@ -0,0 +1,231 @@
+import changedDate from '../../utils/changedDate';
+import changedTime from '../../utils/changedTime';
+
+import getPinterestPinId from './getPinterestPinId';
+import sizeOptions from './sizeOptions';
+
+export default class PinterestPinBlockModal extends Component {
+  static propTypes = {
+    block: PropTypes.instanceOf(Message).isRequired,
+    isFreshBlock: PropTypes.bool.isRequired,
+    isOpen: PropTypes.bool,
+    toggle: PropTypes.func.isRequired,
+    onAddBlock: PropTypes.func.isRequired,
+    onEditBlock: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    isOpen: false,
+  };
+
+  constructor(props) {
+    super(props);
+    const { block } = props;
+
+    this.state = {
+      aside: block.get('aside'),
+      errorMsg: '',
+      hasUpdatedDate: block.has('updated_date'),
+      href: block.get('href'),
+      isValid: block.has('href'),
+      size: block.get('size'),
+      terse: block.get('terse'),
+      touched: false,
+      updatedDate: block.get('updated_date', new Date()),
+    };
+
+    this.handleAddBlock = this.handleAddBlock.bind(this);
+    this.handleChangeCheckbox = this.handleChangeCheckbox.bind(this);
+    this.handleChangeDate = this.handleChangeDate.bind(this);
+    this.handleChangeTextarea = this.handleChangeTextarea.bind(this);
+    this.handleChangeTime = this.handleChangeTime.bind(this);
+    this.handleEditBlock = this.handleEditBlock.bind(this);
+    this.handleChangeSelect = this.handleChangeSelect.bind(this);
+  }
+
+  componentDidMount() {
+    setTimeout((t) => {
+      t.inputElement.focus();
+    }, 0, this);
+  }
+
+  setBlock() {
+    const { hasUpdatedDate, terse, href, size, updatedDate, aside } = this.state;
+    const { block } = this.props;
+    return block.schema().createMessage()
+      .set('aside', aside)
+      .set('href', href)
+      .set('size', size)
+      .set('terse', terse)
+      .set('updated_date', hasUpdatedDate ? updatedDate : null);
+  }
+
+  handleAddBlock() {
+    const { onAddBlock, toggle } = this.props;
+    onAddBlock(this.setBlock());
+    toggle();
+  }
+
+  handleEditBlock() {
+    const { onEditBlock, toggle } = this.props;
+    onEditBlock(this.setBlock());
+    toggle();
+  }
+
+  handleChangeCheckbox({ target: { checked, id } }) {
+    this.setState({ [id]: checked });
+  }
+
+  handleChangeDate(date) {
+    this.setState(changedDate(date));
+  }
+
+  handleChangeTime({ target: { value: time } }) {
+    this.setState(changedTime(time));
+  }
+
+  handleChangeTextarea(event) {
+    let { errorMsg, href, isValid, size, terse } = this.state;
+    const input = event.target.value;
+    const pinId = getPinterestPinId(input);
+
+    const embed = new DOMParser().parseFromString(input, 'text/html');
+    const anchor = embed.querySelector('a');
+    const pinWidth = anchor ? anchor.dataset.pinWidth : '';
+    const pinTerse = anchor ? anchor.dataset.pinTerse : '';
+
+    if (pinId) {
+      errorMsg = '';
+      isValid = true;
+      href = pinId;
```

Comment:
@jalopez13 this seems weird, renaming util to getPinterestPinHref would be clearer

Reply:
@efjacobson 

Renamed to getPinterestPinUrl.js.